### PR TITLE
CLDC-3638 Allow adding custom resources

### PR DESCRIPTION
**Pull request by kosiakkatrina**
Allow adding custom resources, run the same file validations for size and presence, add a resource type and validate that it's present. Also allow updating the resource, the flow for which would work the same as updating mandatory resources
image

### CODE REVIEW

File: app/controllers/collection_resources_controller.rb [hunk 1]
```diff
@@ -23,7 +25,16 @@ def download_mandatory_collection_resource
     download_resource(resource.download_filename)
   end
 
-  def edit
+  def download_additional_collection_resource
```
Comment (RachaelBooth) [inline app/controllers/collection_resources_controller.rb:28]: Does this also need an exception from the `before_action :authenticate_user!`?

File: app/views/collection_resources/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      </p>
+
+      <%= f.govuk_file_field :file,
+        label: { text: "Upload file", size: "m" } %>
+
+      <%= f.govuk_text_field :short_display_name,
+                      label: nil,
+                      hint: { text: safe_join(["This will be used in the download link on the homepage. Do not include the log type or collection year.",
+                                               content_tag(:br),
+                                               "For example, if you enter “bulk upload change log”, the download link will say “Download the #{@collection_resource.log_type} bulk upload change log (#{text_year_range_format(@collection_resource.year)})”."]) } %>
```
Comment (Dinssa) [inline app/views/collection_resources/new.html.erb:28]: Could you add one more `<br>`? On the edit page too. Separate out the sentences more clearly, similar to the designs.

**Review by RachaelBooth — APPROVED** [review]

File: app/views/collection_resources/new.html.erb (hunk 2 above)
Comment (kosiakkatrina) [reply app/views/collection_resources/new.html.erb:28]: This is what it looks like atm, the description screenshot it a bit outdated. What do you think?
image

File: app/controllers/collection_resources_controller.rb (hunk 1 above)
Comment (kosiakkatrina) [reply app/controllers/collection_resources_controller.rb:28]: Yup! Good spot!

File: app/views/collection_resources/new.html.erb (hunk 2 above)
Comment (Dinssa) [reply app/views/collection_resources/new.html.erb:28]: Okay that's all good

**Pull request merged**